crates/ra_ide/src/inlay_hints.rs

```diff
@@ -112,58 +112,73 @@ fn get_param_name_hints(
     // we need args len to determine whether to skip or not the &self parameter
     .collect::<Vec<_>>();
 
-    let (has_self_param, fn_signature) = get_fn_signature(db, analyzer, &expr)?;
-    let parameters = if has_self_param && fn_signature.parameter_names.len() > args.len() {
-        fn_signature.parameter_names.into_iter().skip(1)
-    } else {
-        fn_signature.parameter_names.into_iter().skip(0)
-    };
-
-    let hints =
-        parameters
-            .zip(args)
-            .filter_map(|(param, arg)| {
-                if !param.is_empty() {
-                    Some((arg.syntax().text_range(), param))
-                } else {
-                    None
-                }
-            })
-            .map(|(range, param_name)| InlayHint {
-                range,
-                kind: InlayKind::ParameterHint,
-                label: param_name.into(),
-            });
+    let fn_signature = get_fn_signature(db, analyzer, &expr)?;
+    let parameters =
+        if fn_signature.has_self_param && fn_signature.parameter_names.len() > args.len() {
+            fn_signature.parameter_names.iter().skip(1)
+        } else {
+            fn_signature.parameter_names.iter().skip(0)
```

Comment:
Nifty!

Though, perhaps

```rust
let n_params_to_skip = if has_self { 1 } else { 0}
```

?

lifting common subexrepssions is usually a good idea

Reply:
Sure.
The pattern itself (considering &self param) occurs in this file's logic at least twice, so maybe later it would be good to move the common logic into the `FunctionSignature`.